### ISSUE TITLE
`OfflineAudioContext` support

### ISSUE THREAD

**Issue by mrkev** [Smona/faust-loader#5]
Hello! First and foremost, a huge thanks for working on `faust-loader`. I'm using it for a project and it's great!

I'm wondering what's blocking support for `OfflineAudioContext`. I'm implementing "bouncing" functionality in which I try to render a stream, as quickly as possible, into a buffer, but currently I have to do so bypassing faust effects on that stream because I use `OfflineAudioContext` to achieve it. If it's something I can help with I'd be happy to lend a hand.

**Comment by mrkev** [comment]
I had a quick look and I think I know what's creating at least the immediate exception in my app. There's this cache for modules, so that modules that have already been added aren't added again.

https://github.com/Smona/faust-loader/blob/8c6ff651ee3a6e95dda93f6dba2d219bcc9ee9ad/src/loadProcessor.ts#L21-L30

This cache doesn't consider the context they were added to, just the URL. So it returns a module that has been initialized to another context in my application (since I use a normal `AudioContext` at initialization, but am trying to initialize these modules with in `OfflineAudioContext` afterwards, when the user tries to bounce the tracks).

This explains at least one of the errors I get:

```
loadProcessor.js:144 DOMException: Failed to construct 'AudioWorkletNode': AudioWorkletNode cannot be created: The node name 'Panner' is not defined in AudioWorkletGlobalScope.
```

The node is certainly not defined in `AudioWorkletGlobalScope`, because we skipped that part and returned one that's defined in another audio context!

I'll try to fiddle with this (maybe not today though).

**Comment by mrkev** [comment]
Last update, I ended up commenting out the lines that check the cache real quick:

```typescript
// if (existing) { 
//     return existing; 
// } 
```

No exceptions! The bounce works! Unfortunately it generates a buffer with no audio; it's a completely mute wav. I guess I just confirmed at least API-wise one could swap the `AudioContext` for the `OfflineAudioContext` and get no errors, but getting no audio back seems to be back in the realm of "could `OfflineAudioContext` acutally be _used_ though?". Logging off for real now though. ✌️

**Comment by Smona** [comment]
I had been eyeing the NPM downloads, but I'm elated to hear for certain that someone besides me is getting use out of the package!

I appreciate the investigation; we should definitely be caching per-context as well. The loader should support N contexts, and absolutely should support Faust nodes in an OfflineAudioContext.

I had built this loader as part of a much larger project so I moved on once it reached MVP level, but you've prompted me to take a look at cleaning it up a bit. Right now I'm updating the FaustAudioProcessorNode code to pass type checks and expand the typing for the returned node. After that I'll look at creating a working test case for multiple (possibly offline) audio contexts.

In the meantime, if you're able to get offline rendering audio output working, PRs are absolutely welcome!
